### PR TITLE
Support multiple Facets

### PR DESCRIPTION
**Pull request by goaround**
### Description of the Change

Change `querySelector` to `querySelectorAll` and use a `forEach` to set the event listener to all Facets

### Alternate Designs

None

### Benefits

You can now search more than just on the first Facet Widget for Terms

### Possible Drawbacks

None

### Verification Process

Tried it on my development environment

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

Closes: #2003 

### Changelog Entry

- Fixed Facet Term Search for more than one Widget

### CODE REVIEW

File: assets/js/facets.js [hunk 1]
```diff
@@ -29,17 +27,19 @@ const handleFacetSearch = (event) => {
...
-const facetSearchInput = document.querySelector('.widget_ep-facet .facet-search');
+const facets = document.querySelectorAll('.widget_ep-facet');
 
-if (facetSearchInput) {
-	facetSearchInput.addEventListener(
+facets.forEach(facet => {
+	const facetSearchInput = facet.querySelector('.facet-search');
+	const facetTerms = facet.querySelector('.terms');
+	
+	facet.querySelector('.facet-search').addEventListener(
```
Comment (brandwaffle) [inline assets/js/facets.js:36]: addEventListener is triggering an error for any Facet that does not have a search bar (i.e. only has a few options). We need a conditional to check for that here.

**Comment by brandwaffle** [comment]
This pull request is related to #2041

File: assets/js/facets.js (hunk 1 above)
Comment (goaround) [reply assets/js/facets.js:40]: I add an `if` statement if `facetSearchInput` exists

**Comment by brandwaffle** [comment]
@goaround thank you for contributing this!!! We have it slotted for our next release.

File: assets/js/facets.js [hunk 2]
```diff
@@ -29,17 +27,23 @@ const handleFacetSearch = (event) => {
...
 			if (event.keyCode === 13) {
 				return;
 			}
 
-			handleFacetSearch(event);
-		}, 200),
-	);
-}
+			handleFacetSearch(event, facetTerms);
+		}, 200)
```
Comment (Rahmon) [inline assets/js/facets.js:48]: the function addEventListernet is missing a ")".

File: assets/js/facets.js [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
-
 /**
  * Filters the facets to match the input search term when
  * the number of terms exceeds the threshold determined
  * by the ep_facet_search_threshold filter
  *
  * @param {event} event - keyup
  */
-const handleFacetSearch = (event) => {
+const handleFacetSearch = (event, facetTerms) => {
```
Comment (Rahmon) [inline assets/js/facets.js:10]: the facetTerms parameter is missing the type and description.

**Comment by Rahmon** [comment]
@goaround I've left some comments in the code. Moreover, you should include the facets-script.min.js and facets-script.min.js.map. You can generate those running `npm run build`

Thank you for your contribution and let me know if you have any questions.

File: assets/js/facets.js (hunk 3 above)
Comment (goaround) [reply assets/js/facets.js:10]: Added

File: assets/js/facets.js (hunk 2 above)
Comment (goaround) [reply assets/js/facets.js:48]: Fixed and check if dist file is working

**Comment by goaround** [comment]
Fixed the issue and build the dist files.

**Review by Rahmon — APPROVED** [review]
LGTM :+1:

**Review by brandwaffle — COMMENTED** [review]
LGTM

**Pull request merged**